src/reuse/modules/service/odata.ts

```diff
@@ -55,31 +78,29 @@ export class OData {
     };
 
     const parameters = {
-      "sap-client": "715",
-      "sap-documentation": ["heading", "quickinfo"],
-      "sap-language": "EN"
+      ...params,
+      ...{
+        "sap-client": "715",
+        "sap-documentation": ["heading", "quickinfo"],
+        "sap-language": "EN"
+      }
     };
 
-    if (params) {
-      // @ts-ignore
-      Object.keys(params).forEach((key) => (parameters[key] = params[key]));
-    }
-
     const auth: any = {
+      type: authType ?? "",
       username: username,
       password: password
     };
+    if (headers && Object.entries(headers).length > 0) auth.headers = headers;
 
-    if (authType) {
-      auth["type"] = authType;
-    }
     const srv = new this.Service({
       logger: loggingEnabled ? logger : "",
       url: url,
       auth: auth,
-      parameters: parameters, //Define initial request by $metadata?sap-client=<client-number>&sap-documentation=&sap-language=EN
+      parameters: parameters, // Define initial request by $metadata?sap-client=<client-number>&sap-documentation=&sap-language=EN
```

Comment:
Just a suggestion here, not critical: use the shorthand property assignment, when value for some key inside an object is taken from a variable with the same name as the key has

`const src = ...{
  ...,
  url,
  auth,
  parameters
}`

More info just in case: https://ui.dev/shorthand-properties

Reply:
Sure thing! Will change it, thanks!

Reply:
done